src/components/Container.js

```diff
@@ -96,19 +96,29 @@ class Container extends React.PureComponent {
           progress={{ id: this.state.currentId, completed: this.state.count / ((this.props.stories[this.state.currentId] && this.props.stories[this.state.currentId].duration) || this.defaultInterval) }}
           progressAtBottom={this.props.progressAtBottom}
         />
-        <Story
-          ref={s => this.story = s}
-          action={this.pause}
-          bufferAction={this.state.bufferAction}
-          height={this.height}
-          playState={this.state.pause}
-          width={this.width}
-          story={this.props.stories[this.state.currentId]}
-          loader={this.props.loader}
-          header={this.props.header}
-          getVideoDuration={this.getVideoDuration}
-          storyContentStyles={this.props.storyContentStyles}
-        />
+
+        {this.props.stories.map((story, storyIndex) => {
+          const active = this.state.currentId === storyIndex
+
+          return (
+            <Story
+              ref={s => this.story = s}
```

Comment:
You can move this function to a handle and call it here. Doing this, this function will be re-rendered every time and it will create a new instance of it